### ISSUE TITLE
JupyterLab O/S' launched by JupyterHub (i.e. by DockerSpawner) is causing cgroup-related issues such that podman(1) fails ...

### ISSUE THREAD

**Issue by nmvega** [jupyterhub/dockerspawner#402]
Dear Friends:

I provide a `very detailed` background on this issue [here](https://github.com/jupyterhub/jupyterhub/issues/3309), but that issue is closed (wrong repo) and I need assistance. Please read that question because it is what this issue here is about. (Thank you).

When `JupyterHub` spawns (via `DockerSpawner`) a `JupyterLab` container image, it is missing `docker` runtime attributes and/or `O/S capabilities` that allow RedHat's `podman(1)` to work properly. Here's a summary example, but complete information is available at the above URL:

```
jovyan@lCONTAINER$ mkdir -p ${HOME}/data.d/mariadb.d/var/lib/mysql

jovyan@CONTAINER$ podman run --events-backend=file --name mariadb01 \
   -p 3306:3306 -v ${HOME}/data.d/mariadb.d/var/lib/mysql:/var/lib/mysql:Z \
   -e MYSQL_ROOT_PASSWORD=**** -e MYSQL_DATABASE=db01 \
   -e MYSQL_USER=jdoe -e MYSQL_PASSWORD=**** -d docker.io/library/mariadb

Error: OCI runtime error: container_linux.go:370: starting container process caused:
process_linux.go:326: applying cgroup configuration for process caused:
no cgroup mount found in mountinfo
```

The above is **not** an issue when I spawn the identical container-image manually (again please see URL above for blow-by-blow walk-through), but it is a problem when `JupyterHub` spawns it via `DockerSpawner`. Please help me.

Thank you!

**Comment by consideRatio** [comment]
@nmvega, I believe it will be an issue for the DockerSpawner repo, do you agree? JupyterHub doesn't know anything about containers.

**Comment by nmvega** [comment]
@consideRatio Yes, I agree. That makes sense. I can close this issue with a note (_that this wasn't the optimal repo for it_ ) and repost the issue at the `DockerSpawner` repo? I only ask because I don't know if GitHub has a `move` tool. I doubt it. 😆

**Comment by consideRatio** [comment]
Hehe it has and i tranferred it :) It is already in the dockerspawner repo. It requires you to have rights in both repos to do it though.

**Comment by nmvega** [comment]
> Hehe it has and i tranferred it :) It is already in the dockerspawner repo. It requires you to have rights in both repos to do it though.

Thank you! I'll also modify the post's wording to reference `DockerSpawner` because (as you pointed out), that's a better way to frame this issue.

**Comment by nmvega** [comment]
Hi again @consideRatio:

Would you like to work with me to prune some of the comments and make this issue leaner? I was thinking of keeping just these two comments, and removing the others:

- [Keep this Original Post.](https://github.com/jupyterhub/dockerspawner/issues/402#issue-770886648)
- [And keep this follow-up comment, where I mention you.](https://github.com/jupyterhub/dockerspawner/issues/402#issuecomment-747723211)

My fear is that the thread will lose it's gist, and people will wander away. LoL =:) What do you think? 
If you agree, you can delete the other comments, including this one (leaving only the two listed above). You can email me at nmvega AT prismalytics DOT io, if you need. Thank you.

**Comment by meeseeksmachine** [comment]
This issue has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/podman-1-not-working-inside-jupyterlab-containers-that-are-launched-by-jupyterhub/7188/4

**Comment by consideRatio** [comment]
@nmvega this has already taken quite a bit of my attention, which is something I'd really prefer to direct on working to improve the open source project i care for more directly.

As communicated through [this message](https://github.com/jupyterhub/jupyterhub/issues/3309#issuecomment-747237435), we wish to separate support/requests for help from clearly defined problems with the projects content. This separation isn't black and white, its often grey.

In my mind, your issue is more a request for help rather than a request for something to be changed. Your request for help is to understand something of relevance, that in turn could lead to a request to change something though, that makes it less clear cut. I hope you can sympathize with the challenge for the volunteer maintainers of this open source project to manage this balance with the limited time we put in.

At this point, I ask that anyone wanting to reason about this, focus the discussion towards [the forum post](https://discourse.jupyter.org/t/podman-1-not-working-inside-jupyterlab-containers-that-are-launched-by-jupyterhub/7188/4) you have made until a clear suggestion for a GitHub repo content change emerge.

**Comment by nmvega** [comment]
@consideRatio 
Okay, though I only requested the deletion of a few comments, so others wouldn't be distracted by the unnecessary `kubernetes` stuff. Nothing more.

As to the core issue, at [this point](https://discourse.jupyter.org/t/podman-1-not-working-inside-jupyterlab-containers-that-are-launched-by-jupyterhub/7188/6?u=nmvega) in the discussion, I'm nearly out of ideas (_having tried numerous variations_), since I don't know what `DockerSpawner` (via `./jupyterhub_config.py`) requires to make both scenarios work the same.

I'm currently comparing (programmatically) `JSON` outputs that I affixed to that discussion. Although when I identify the material differences, I won't know what the `./jupyterhub_config.py` attributes, data structures and values should be (... not initially. I'll have to compare [this low-level docker API](https://docker-py.readthedocs.io/en/stable/api.html) with [this and other JupyterHub APIs](https://jupyterhub-dockerspawner.readthedocs.io/en/latest/api/) and try different things).
